cmd/juju/controller/listmodels.go

```diff
@@ -81,6 +83,18 @@ func (c *modelsCommand) SetFlags(f *gnuflag.FlagSet) {
 
 // Run implements Command.Run
 func (c *modelsCommand) Run(ctx *cmd.Context) error {
+	ctrlRoot, err := c.NewAPIRoot()
```

Comment:
I think we should be returning the is-controller flag back in the API call when asking for the model list.

Otherwise other API users will need to duplicate this logic.